scipy/interpolate/_bary_rational.py

```diff
@@ -560,5 +561,115 @@ def clean_up(self, cleanup_tol=1e-13):
+    Barycentric rational interpolant with no poles on the real axis.
+
+    Parameters
+    ----------
+    points : 1D array_like, shape (n,)
+        1-D array containing values of the independent variable. Values may be real or
+        complex but must be finite.
+    values : 1D array_like, shape (n,)
+        Function values ``f(z)`` at `points`. Infinite and NaN values of `values` and
+        corresponding values of `points` will be discarded.
+    d : int, optional
+        Blending parameter which controls the blending of interpolating polynomials, see
+        notes for more details. For ``d=n-1`` it is equivalent to polynomial
+        interpolation. Must satisfy ``0 <= d < n``, defaults to 3.
+
+    Attributes
+    ----------
+    support_points : array
+        Support points of the interpolant. These are the provided `points`.
+    support_values : array
+        Value of the `interpolant` at the `support_points`.
+    weights : array
+        Weights of the barycentric approximation.
+
+    See Also
+    --------
+    AAA : Barycentric rational approximation of real and complex functions.
+    pade : Padé approximation.
+
+    Notes
+    -----
+    The Floater-Hormann interpolant is a rational function that interpolates the data
+    with approximation order :math:`O(h^{d+1})`. The rational function contains no poles
+    on the real axis, unlike `AAA`. The interpolant is given
+    by
+
+    .. math::
+
+        r(z) = \frac{\sum_{i=0}^{n-d} \lambda_i(z) p_i(z)}
+        {\sum_{i=0}^{n-d} \lambda_i(z)},
+
+    where :math:`p_i(z)` are unique interpolating polynomials of at most degree `d` of
+    the `points` and `values` :math:`(x_i,f_i),\dots,(x_{i+d},f_{i+d}), and
+    :math:`\lambda_i(z)` are blending functions. When ``d = n - 1`` this reduces to
+    polynomial interpolation.
+
+    In practice, the following barycentric representation is used instead
+
+    .. math::
+
+        r(z) = \frac{\sum_{k=1}^m\ w_k f_k / (z - z_k)}{\sum_{k=1}^m w_k / (z - z_k)},
+
+    where the weights :math:`w_j` are computed as
+
+    .. math::
+
+        w_k &= (-1)^{k - d} \sum_{i \in J_k} \prod_{j = i, j \neq k}^{i + d}
+        1/|x_k - x_j|, \\
+        J_k &= \{ i \in I: k - d \leq i \leq k\},\\
+        I &= \{0, 1, \dots, n - d\}.
+
+    References
+    ----------
+    .. [1] M.S. Floater and K. Hormann, "Barycentric rational interpolation with no
+           poles and high rates of approximation", Numer. Math. 107, 315 (2007).
+           :doi:`10.1007/s00211-007-0093-y`
+
+    Examples
+    --------
+
+    Here we compare the method against polynomial interpolation for an example where
+    the polynomial interpolation fails due to Runge's phenomenon.
+
+    >>> import numpy as np
+    >>> from scipy.interpolate import FloaterHormann, BarycentricInterpolator
+    >>> def f(z):
+    ...     return 1/(1 + z**2)
+    >>> z = np.linspace(-5, 5, num=15)
+    >>> r = FloaterHormann(z, f(z))
+    >>> p = BarycentricInterpolator(z, f(z))
```

Comment:
Minor, optional: do we want to advertise BarycentricInterpolator in the docs?  Consider doing a naive thing with a det(Vandermonde matrix) --- fine by me not to if you've a strong opinion though

Reply:
I'd personally prefer to advertise rather than do something ad hoc